src/PlatformEngines.jl

```diff
@@ -630,25 +718,32 @@ Unpack tarball located at file `tarball_path` into directory `dest`.
 """
 function unpack(tarball_path::AbstractString, dest::AbstractString;
                 verbose::Bool = false)
-    # The user can force usage of our dereferencing workaround for filesystems
+
+    # unpack into dest
+    mkpath(dest)
+
+    # The user can force usage of our dereferencing workarounds for filesystems
     # that don't support symlinks, but it is also autodetected.
-    copyderef = get(ENV, "BINARYPROVIDER_COPYDEREF", "") == "true" ||
-                (tempdir_symlink_creation && !probe_symlink_creation(dest))
-
-    # If we should "copyderef" what we do is to unpack into a temporary directory,
-    # then copy without preserving symlinks into the destination directory.  This
-    # is to work around filesystems that are mounted (such as SMBFS filesystems)
-    # that do not support symlinks.  Note that this does not work if you are on
-    # a system that completely disallows symlinks (Even within temporary
-    # directories) such as Windows XP/7.
-    true_dest = dest
+    copyderef = (get(ENV, "BINARYPROVIDER_COPYDEREF", "") == "true") || !probe_symlink_creation(dest)
+
+    # If we should "copyderef" what we do is to unpack everything except symlinks
+    # then copy the sources of the symlinks to the destination of the symlink instead.
+    # This is to work around filesystems that are mounted (such as SMBFS filesystems)
+    # that do not support symlinks.
+
+    excludelist = ""
```

Comment:
default `excludelist` to `nothing`

Reply:
I digged a bit into interpolation from commands and I think I now understand the behavior:
```
x = ``
`command -o$x` # will evaluate to `command` and drop the option completely
x = "my file"
`command -o$x` # will evaluate to `command '-omy file'`
```
So what about defaulting ```excludelist::Union{AbstractString, Cmd} = `` ```
Then we can skip the `excludelist == nothing ? ... : ...` and simply use
```
        pipeline(`$exe7z x $(tarball_path) -y -so`,
                 `$exe7z x -si -y -ttar -o$(out_path)  -x@$(excludelist)`)
```
and
```
return `$tar_cmd -x$(Jjz)f $(tarball_path) --directory=$(out_path) --exclude-from=$(excludelist)`
```
I rewrote it and it workslike a charm ...

Reply:
So what's happening here is a kind of surprising interaction in the `Cmd` parser where it's seeing `command -o$x` as a cartesian product between arrays to be expanded, as described [in the docs here](https://docs.julialang.org/en/v1/manual/running-external-programs/index.html#command-interpolation-1).  Essentially, it looks at the `Cmd` like a container, and tries to iterate over it, but the empty `Cmd` is seen as a zero-dimensional object, and this disables all iteration, so we never see the first part (the `"-o"`).

This is not very intuitive at all, and I would not be surprised if this changes in a future version of Julia, so I think it's best not to rely on it.  Let's not use `Cmd` objects at all when interpolating into other `Cmd` objects and only use Strings; that's what is documented best, and is least likely to change in a future version.

Using `nothing` as the sentinel value for "do nothing" is pretty idiomatic, so I think thats better than using something else like a `Cmd` or an empty string; it fits in better with the rest of the APIs in Julia.

Thanks for putting so much work into this!